src/qgraphicsmozview_p.cpp

```diff
@@ -425,6 +431,33 @@ void QGraphicsMozViewPrivate::UpdateViewSize()
         return;
     }
 
+    if (mOrientationDirty) {
+        QMatrix affine;
+        ScreenRotation rotation = ROTATION_0;
+        switch (mOrientation) {
+        case Qt::LandscapeOrientation:
+            rotation = mozilla::ROTATION_90;
+            affine.rotate(90);
+            affine.translate(0, -mGLSurfaceSize.height());
+            break;
+        case Qt::InvertedLandscapeOrientation:
+            rotation = mozilla::ROTATION_270;
+            affine = affine.rotate(270);
+            affine.translate(-mGLSurfaceSize.width(), 0);
+            break;
+        case Qt::InvertedPortraitOrientation:
+            rotation = mozilla::ROTATION_180;
+            affine = affine.rotate(180);
+            affine.translate(-mGLSurfaceSize.width(), -mGLSurfaceSize.height());
+            break;
+        default:
+            break;
+        }
+        gfxMatrix matr(affine.m11(), affine.m12(), affine.m21(), affine.m22(), affine.dx(), affine.dy());
+        mView->SetScreenRotation(rotation, matr);
```

Comment:
I agree that it'd should be like SetScreenRotation(mozilla::ScreenRotation rotation). Problem is that we need to take dimensions of the new orientation into account so that translation matrix is calculated correctly. This means that call order matters but in away that could be acceptable.

So this would look like:
```
    if (mContext->GetApp()->IsAccelerated() && mHasContext) {
        mView->SetGLViewPortSize(mGLSurfaceSize.width(), mGLSurfaceSize.height());
    }
    mView->SetViewSize(mSize.width(), mSize.height());

    if (mOrientationDirty) {
        ScreenRotation rotation = ROTATION_0;
        switch (mOrientation) {
        case Qt::LandscapeOrientation:
            rotation = mozilla::ROTATION_90;
            break;
        case Qt::InvertedLandscapeOrientation:
            rotation = mozilla::ROTATION_270;
            break;
        case Qt::InvertedPortraitOrientation:
            rotation = mozilla::ROTATION_180;
            break;
        default:
            break;
        }
        mView->SetScreenRotation(rotation);
        mOrientationDirty = false;
    }
```
Much better form API point of view and later we can do needed orientation related handling in embedlite.

What do you think?